File: common/src/main/java/com/leo/commonutil/app/ContextHelp.java
package com.leo.commonutil.app;

import android.content.Context;

import java.lang.ref.WeakReference;

public class ContextHelp {
    private static WeakReference<Context> weakReference;

    private ContextHelp() {
    }

    /**
     * 必须在Application中调用
     *
     * @param context
     */
    public static void setContext(Context context) {
        ContextHelp.weakReference = new WeakReference<>(context.getApplicationContext());
    }

    public static Context getContext() {
        if (null == weakReference) {
            throw new RuntimeException("must setContext() in application first");
        }
        Context context = weakReference.get();
        if (null == context) {
            throw new RuntimeException("must setContext() in application first");
        }
        return context;
    }
}

File: system/src/main/java/com/leo/system/context/ContextHelp.kt
package com.leo.system.context

import android.content.Context
import java.lang.ref.WeakReference

object ContextHelp {
    private var weakReference: WeakReference<Context>? = null

    var context: Context
        get() {
            if (null == weakReference) {
                throw RuntimeException("must setContext() in application first")
            }
            return weakReference!!.get() ?: throw RuntimeException("must setContext() in application first")
        }

        set(context) {
            weakReference = WeakReference(context.applicationContext)
        }
}
